File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phoneBook) {
    const phoneSet = new Set(phoneBook);
    
    for (const number of phoneBook) {
        for (let i = 1; i < number.length; i++) {
            const prefix = number.slice(0, i);
            if (phoneSet.has(prefix)) {
                return false;
            }
        }
    }
    return true;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
'''
def solution(phone_book):
    phone_book.sort()
    length = len(phone_book)
    # 23, 88, 123, 567, 2351
    for i in range(length-1):
        for j in range(i+1,length):
            pi = phone_book[i]
            pj = phone_book[j]
            if len(pj) >= len(pi) and pj.startswith(pi):
                return False
    return True
    '''
def solution(phone_book):
    answer = True
    phone_book.sort()
    for i in range(len(phone_book) - 1) :
        if phone_book[i+1].startswith(phone_book[i]) : return False
    return answer